iwyu.cc

```diff
@@ -1751,22 +1752,12 @@ class IwyuBaseAstVisitor : public BaseAstVisitor<Derived> {
   // itself can be forward-declared, just as in (2).  That is:
   //   template<class T> struct C { typedef pair<T,T> value_type; };
   // iwyu will demand the full type of pair, but not of its template
-  // arguments.  This is handled not here, but below, in
-  // VisitSubstTemplateTypeParmType.
+  // arguments.  This is handled in VisitSubstTemplateTypeParmType.
+  // Typedef underlying type is marked fwd-declarable by default, and the
+  // aforementioned analysis is performed with 'IsProvidedComponent' function.
   bool VisitTypedefNameDecl(clang::TypedefNameDecl* decl) {
-    if (CanIgnoreCurrentASTNode())
-      return true;
-    const Type* underlying_type = decl->getUnderlyingType().getTypePtr();
-    const Type* deref_type =
-        RemovePointersAndReferencesAsWritten(underlying_type);
-
-    if (CodeAuthorWantsJustAForwardDeclare(deref_type, GetLocation(decl)) ||
-        isa<SubstTemplateTypeParmType>(underlying_type)) {
-      current_ast_node()->set_in_forward_declare_context(true);
-    } else {
-      current_ast_node()->set_in_forward_declare_context(false);
-    }
-
+    if (CanIgnoreCurrentASTNode())  return true;
```

Comment:
This looks copy-pasted from before 890a3150d74f834a31e57c649195d6d68da682b1. Let's clang-format the line.